src/IdentityServer4/Hosting/CookieMiddleware.cs

```diff
@@ -30,6 +30,9 @@ public static void ConfigureCookies(this IApplicationBuilder app)
                         SlidingExpiration = options.Authentication.CookieSlidingExpiration,
                         ExpireTimeSpan = options.Authentication.CookieLifetime,
                         CookieName = IdentityServerConstants.DefaultCookieAuthenticationScheme,
+                        LoginPath = options.UserInteraction.LoginUrl,
```

Comment:
you'd want something like this:
```
if (url.IsLocalUrl())
{
    if (url.StartsWith("~/")) url = url.Substring(1);
}
```

and then assign the url to the `LoginPath` and/or `LogoutPath`. If they're not local, then don't assign them.